client/my-sites/plans-features-main/hooks/use-is-plan-upsell-enabled-on-free-domain.ts

```diff
@@ -0,0 +1,23 @@
+import { useExperiment } from 'calypso/lib/explat';
+import type { DataResponse } from 'calypso/my-sites/plan-features-2023-grid/types';
+
+export const useIsPlanUpsellEnabledOnFreeDomain = (
+	flowName?: string | null
+): DataResponse< boolean > => {
+	const ONBOARDING_EXPERIMENT = 'calypso_gf_signup_onboarding_free_free_dont_miss_out_modal';
+	const ONBOARDING_PM_EXPERIMENT = 'calypso_gf_signup_onboarding_pm_free_free_dont_miss_out_modal';
+	const relevantExperiment =
+		flowName === 'onboarding' ? ONBOARDING_EXPERIMENT : ONBOARDING_PM_EXPERIMENT;
+	const [ isLoading, experimentAssignment ] = useExperiment( relevantExperiment );
+
+	if ( ! [ 'onboarding', 'onboarding-pm' ].includes( flowName ?? '' ) ) {
```

Comment:
This works, but let's use the `isEligible` functionality of `useExperiment` instead of redone it here. e.g. https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plans-features-main/hooks/use-is-custom-domain-allowed-on-free-plan.ts#L13

That way this function will be simpler and the return value will stay the same.